tests/src/test/java/alluxio/server/ft/ZookeeperFailureIntegrationTest.java

```diff
@@ -82,7 +84,13 @@ public void zkFailure() throws Exception {
     CommonUtils.waitFor("a successful operation to be performed", () -> thread.successes() > 0);
     mCluster.stopZk();
     long zkStopTime = System.currentTimeMillis();
-    CommonUtils.waitFor("operations to start failing", () -> thread.getLatestFailure() != null);
+    // Wait until 3 different failures are encountered on the thread.
+    // PS: First failures could be related to worker capacity depending on process shutdown order,
+    // thus still leaving RPC server reachable.
+    AtomicInteger failureCounter = new AtomicInteger(3);
```

Comment:
Why did this fail for the PR?

Reply:
Don't really know. My guess is new streaming version of call introduced a timing condition which surfaced this issue.